dts/bindings/serial/ite,it8xxx2-uart.yaml

```diff
@@ -0,0 +1,18 @@
+# Copyright (c) 2020 ITE Corporation. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+description: ITE it8xxx2 ns16550 UART
```

Comment:
If the uart is ns16550 than can we not utilize that existing binding and compatible for that?

Reply:
The it8xxx2 chip is based on ns16550, but there are still some differents.
Such as:
a. The it8xxx2 not support RXRDY and TXRDY functions
b. The it8xxx2 support DMA mode
c. The it8xxx2 have to initial uart module at begin.
...etc.
If binding them, we are afraid of causing the ns16550 code confusion and unreadable.

Reply:
what do you mean next version?

Reply:
Hi We'v revised our code, it become using the ns16550 seial driver, and removed our uart driver.
The uart hw setting were moved to pinmux.c and soc.c  
Please review it again, thank you.